### PR TITLE
Import changes from old fork

### PR DESCRIPTION
**Pull request by M4LuZ**
### What is this PR doing?

There are multiple smaller fixes and improvements (and also some larger ones) that I've developed in the past in a fork.
As these are quite outdated I'll fix, refactor, improve as needed.
This is the first batch as foundation to then implement some larger and more interesting ones (as support for GiroCode QR-Code creation)

### Which issue(s) this PR fixes:

Fixes #none

### Checklist

- [x] `CHANGELOG.md` entry
- [x] Documentation update

### CODE REVIEW

File: inc/Classes/Display.php [hunk 1]
```diff
@@ -1292,7 +1292,7 @@ public function FetchUserIcon($userid, $username = '')
             $username = '<i>System</i>';
         }
 
-        $smarty->assign('userid', $userid);
+        $smarty->assign('userid', htmlspecialchars($userid));
```
Comment (andygrunwald) [inline inc/Classes/Display.php:1295]: Isn't this always an integer?

File: inc/Classes/Func.php [hunk 2]
```diff
@@ -1118,4 +1118,56 @@ public function isModActive($mod, &$caption = '')
...
+     */
+    public function replaceVariables($text) {
+        global $auth;
+
+        //initialize replacement array
+        $placeholderNames = [];
+        $replacementValues = [];
+
+            $placeholderNames []= '%TEST%';
+            $replacementValues []= 'Test OK!';
```
Comment (andygrunwald) [inline inc/Classes/Func.php:1141]: Is %TEST% really needed?

File: modules/party/Classes/Party.php [hunk 3]
```diff
@@ -390,37 +399,51 @@ public function delete_usergroups($del_group, $set_group)
...
      * Returns the amount of users registered for a party.
-     * 
-     * @param int $party_id The ID of the party to calculate this for
+     *
+     * @param int $partyId The ID of the party to calculate this for (uses object value otherwise)
+     * @param
      * @return array Result array with elements "qty" and "paid"
     */
-    public function getGuestQty($party_id = NULL)
+    public function getGuestQty($partyId = null, $showOrga = null)
```
Comment (andygrunwald) [inline modules/party/Classes/Party.php:410]: Did we ever use the second parameter `$showOrga`?

File: website/docs/modules/Info2.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+sidebar_position: 1
+---
+
+## Module Description
+
+This module provides the abilitity to create and manage multiple rich-text pages for user information pages
+
+## Configuration options
+
+TBD
```
Comment (andygrunwald) [inline website/docs/modules/Info2.md:13]: Is this still work in progress?

**Review by andygrunwald — CHANGES_REQUESTED** [review]
I left a few comments

File: inc/Classes/Display.php (hunk 1 above)
Comment (M4LuZ) [reply inc/Classes/Display.php:1295]: This change was copied over by hand and I added it into the wrong line.
`$userid` is an integer, `$username` is (where it belongs) is a string with potentially html characters.
Will be changed.

File: inc/Classes/Func.php (hunk 2 above)
Comment (M4LuZ) [reply inc/Classes/Func.php:1141]: Nope. But it also does not hurt ;)
Fine to remove

File: website/docs/modules/Info2.md (hunk 4 above)
Comment (M4LuZ) [reply website/docs/modules/Info2.md:13]: No I considered it really to be done in the future when also the other issues of the Info2 module are to be touched or we tackle generic module documentation.

I'm fine to remove the block if that is better?

Comment (andygrunwald) [reply website/docs/modules/Info2.md:13]: Yeah, lets remove it.

Comment (M4LuZ) [reply website/docs/modules/Info2.md]: I've decided to spend the few minutes to document it instead and noticed that option "extended toolbar" is not being used anywhere in the code, thus removed the option and documented the rest

File: modules/party/Classes/Party.php (hunk 3 above)
Comment (M4LuZ) [reply modules/party/Classes/Party.php:410]: This _will_ be used, but is not at this point in time.
There is an config option to include registrations from admins/superadmins in the overview and I've added this to ensure that the function covers both cases. 
This is inteded to replace multiple calculations of similar type (if I remember correctly). 
If it turns out after that that it is not needed, then that's surely fine to remove

**Comment by M4LuZ** [comment]
Should be fine to give it another look

**Review by andygrunwald — APPROVED** [review]
Looking good. Thank you!

**Pull request merged**